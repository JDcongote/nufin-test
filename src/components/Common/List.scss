@import '../../styles/variables/index';
@import'../../styles/tools/mixins';

.list {
  overflow-y: auto;
  flex: 1;
  list-style: none;

  @include breakpoint(tablet-portrait-up) {
    display: grid;
    grid-template-columns: 50% 50%;
  }

  @include breakpoint(tablet-landscape-up) {
    display: grid;
    grid-template-columns: 33.3% 33.3% 33.3%;
    margin-top: 1em;
  }

  @include breakpoint(desktop-up) {
    display: grid;
    grid-template-columns: 25% 25% 25% 25%;
    margin-top: 2em;
  }
}