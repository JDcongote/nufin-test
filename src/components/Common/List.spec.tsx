import List from './list';
import React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';

import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

describe('List', () => {
  it('Basic List test, only items', () => {
    const texts = ['a', 'b'];
    const items = [
      {
        key: 1,
        fragment: <div>a</div>
      },
      {
        key: 2,
        fragment: <div>b</div>
      }
    ];
    const wrapper = mount(<List items={items} />);
    const render = renderer.create(<List items={items} />);
    const nodes = wrapper.find('.list__item');
    nodes.forEach((node, i) => expect(node.text()).toBe(texts[i]));

    expect(render.toJSON()).toMatchSnapshot();
  });

  it('Basic List test, no items', () => {
    const items = [];
    const wrapper = mount(<List items={items} />);
    const nodes = wrapper.find('.list__item');
    expect(nodes.length).toBe(0);
  });
});
