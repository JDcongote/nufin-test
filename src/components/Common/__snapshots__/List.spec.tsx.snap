// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List Basic List test, only items 1`] = `
<ul
  className="list"
>
  <li
    className="list__item"
  >
    <div>
      a
    </div>
  </li>
  <li
    className="list__item"
  >
    <div>
      b
    </div>
  </li>
</ul>
`;
